File: clients/r/fsda_client.R
library(httr)
library(jsonlite)

FSDA_call <- function(func_name, ..., base_url = "http://127.0.0.1:8000") {
  args_list <- list(...)
  
  response <- POST(
    url = paste0(base_url, "/call/", func_name),
    body = toJSON(list(args = args_list), auto_unbox = TRUE),
    content_type_json()
  )
  
  parsed <- content(response, as = "parsed", type = "application/json")
  
  if (status_code(response) != 200) {
    stop(paste("Error:", parsed$detail))
  }
  
  return(parsed$result)
}

File: clients/julia/fsda_client.jl
using HTTP
using JSON

function fsda_call(func_name::String, args...; base_url="http://127.0.0.1:8000")
    body = JSON.json(Dict("args" => collect(args)))
    
    response = HTTP.post(
        "$base_url/call/$func_name",
        ["Content-Type" => "application/json"],
        body
    )
    
    parsed = JSON.parse(String(response.body))
    
    if response.status != 200
        error(parsed["detail"])
    end
    
    return parsed["result"]
end